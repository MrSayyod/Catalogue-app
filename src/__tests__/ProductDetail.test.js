// import React from "react";
// import { act } from "react-dom/test-utils";
// import ProductDetail from "../components/ProductDetail";
// import { render, unmountComponentAtNode } from "react-dom";
// import { Provider } from 'react-redux'
// import store from '../store'

// let container = null;
// beforeEach(() => {
//   container = document.createElement("div");
//   document.body.appendChild(container);
// });

// afterEach(() => {
//   unmountComponentAtNode(container);
//   container.remove();
//   container = null;
// });

// it("renders product data", async() => {
//   const fakeProduct = {
//     id: 1,
//     title: "Backpack",
//     price: 59.99,
//     description: "Perfect pack for everyday use ",
//     category: "men's clothing",
//   };
//   jest.spyOn(global, "fetch").mockImplementation(() =>
//     Promise.resolve({
//       json: () => Promise.resolve(fakeProduct),
//     })
//   );

//   await act(async () => {
//     render(<Provider store={store}><ProductDetail id="1" /></Provider>, container);
//   });

//   expect(container.querySelector("h1").textContent).toBe(fakeProduct.title);
//   expect(container.querySelector("h3").textContent).toBe(fakeProduct.category);
//   expect(container.querySelector("p").description).toBe(fakeProduct.description);

//   global.fetch.mockRestore();
// });
