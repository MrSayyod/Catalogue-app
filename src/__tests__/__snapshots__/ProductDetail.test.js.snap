// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render ProductDetail component 1`] = `[Function]`;
